be/src/storage/rowset_update_state.cpp

```diff
@@ -247,20 +247,103 @@ Status RowsetUpdateState::_prepare_partial_update_states(Tablet* tablet, Rowset*
+        uint32_t num_rows = new_rss_rowids[i].size();
+        std::vector<uint32_t> conflict_idxes;
+        std::vector<uint64_t> conflict_rowids;
+        DCHECK_EQ(num_rows, _partial_update_states[i].src_rss_rowids.size());
+        for (size_t j = 0; j < new_rss_rowids[i].size(); ++j) {
+            uint64_t new_rss_rowid = new_rss_rowids[i][j];
+            uint32_t new_rssid = new_rss_rowid >> 32;
+            uint64_t rss_rowid = _partial_update_states[i].src_rss_rowids[j];
+            uint32_t rssid = rss_rowid >> 32;
+
+            if (rssid != new_rssid) {
+                conflict_idxes.emplace_back(j);
+                conflict_rowids.emplace_back(new_rss_rowid);
+            }
+        }
+        if (!conflict_idxes.empty()) {
+            std::vector<std::unique_ptr<vectorized::Column>> read_columns;
+            std::vector<std::unique_ptr<vectorized::Column>> new_write_columns;
+            read_columns.resize(_partial_update_states[i].write_columns.size());
+            new_write_columns.resize(_partial_update_states[i].write_columns.size());
+            for (uint32_t j = 0; j < new_write_columns.size(); ++j) {
+                read_columns[j] = _partial_update_states[i].write_columns[j]->clone_empty();
+                new_write_columns[j] = _partial_update_states[i].write_columns[j]->clone_empty();
+            }
+            size_t num_default = 0;
+            std::map<uint32_t, std::vector<uint32_t>> rowids_by_rssid;
+            std::vector<uint32_t> read_idxes;
+            plan_read_by_rssid(conflict_rowids, &num_default, &rowids_by_rssid, &read_idxes);
+            DCHECK_EQ(conflict_idxes.size(), read_idxes.size());
+            RETURN_IF_ERROR(tablet->updates()->get_column_values(read_column_ids, num_default > 0, rowids_by_rssid,
+                                                                 &read_columns));
+
+            for (size_t col_idx = 0; col_idx < read_column_ids.size(); col_idx++) {
+                new_write_columns[col_idx]->append_selective(*read_columns[col_idx], read_idxes.data(), 0,
+                                                             read_idxes.size());
+            }
+
+            for (size_t col_idx = 0; col_idx < _partial_update_states[i].write_columns.size(); col_idx++) {
+                RETURN_IF_ERROR(_partial_update_states[i].write_columns[col_idx]->update_rows(
+                        *new_write_columns[col_idx], conflict_idxes.data()));
+            }
+        }
+        int64_t t_resolve_conflict_end = MonotonicMillis();
+        LOG(INFO) << Substitute(
+                "check partial rowset conflict tablet:$0 rowset:$1 seg:$2 #column:$3 #conflict_rows:$4 getrowid:$5ms "
+                "#resolve_conflict $6ms",
+                tablet->tablet_id(), rowset_id, i, read_column_ids.size(), conflict_idxes.size(),
+                (t_get_rowids - t_start) / 1000000, (t_resolve_conflict_end - t_resolve_conflict_start) / 1000000);
```

Comment:
t_get_rowids - t_start  are all the same, maybe we can gather all segments' information and print log after all segments are processed(out of the loop)

Reply:
> t_get_rowids - t_start are all the same, maybe we can gather all segments' information and print log after all segments are processed(out of the loop)

done, change in https://github.com/StarRocks/starrocks/pull/2919/commits/0d57187a8d7ba3a80d526c346f5ff7d10704559d